dial.go

```diff
@@ -93,61 +98,71 @@ func (d *Dialer) Dial(ctx context.Context, raddr ma.Multiaddr, remote peer.ID) (
 			}
 		}()
 
-		maconn, err := d.rawConnDial(ctx, raddr, remote)
+		tptConn, err := d.rawConnDial(ctx, raddr, remote)
 		if err != nil {
 			errOut = err
 			return
 		}
 
 		if d.Protector != nil {
-			pconn, err := d.Protector.Protect(maconn)
+			var pconn tpt.Conn
+			pconn, err = d.Protector.Protect(tptConn)
```

Comment:
The protector should work fine.
`tptConn` is the general `tpt.Conn`. `pnet.Protect` does another type assertion in https://github.com/marten-seemann/go-libp2p-pnet/blob/11a8850893ddc8d571abe57b5082d0110b77d412/protector.go#L33-L42. For `MultiStreamConn`s, it wraps `OpenStream` and `AcceptStream` and uses the PSK with a separate nonce for every stream (see https://github.com/marten-seemann/go-libp2p-pnet/blob/11a8850893ddc8d571abe57b5082d0110b77d412/psk_multistreamconn.go#L22-L36).